packages/api/src/command/medical/patient/get-patient.ts

```diff
@@ -69,54 +88,48 @@ export const getPatientByDemo = async ({
   facilityId: string;
   cxId: string;
   demo: PatientData;
-}): Promise<Patient | null> => {
-  const { log } = Util.out(`getPatientByDemo - cxId ${cxId}`);
+}): Promise<Patient | undefined> => {
+  // Normalize the patient demographic data
+  const normalizedPatientDemo = normalizePatientDataMPI(convertPatientDataToPatientDataMPI(demo));
+  if (!normalizedPatientDemo) {
+    return undefined;
+  }
 
-  const patients = await PatientModel.findAll({
-    where: {
-      cxId,
-      facilityIds: {
-        [Op.contains]: [facilityId],
-      },
-      data: {
-        dob: demo.dob,
-        genderAtBirth: demo.genderAtBirth,
-      },
+  // Block patients based on the criteria of matching dob and genderAtBirth
+  const blockedPatients = await blockPatients({
+    cxId,
+    facilityIds: [facilityId],
+    data: {
+      dob: normalizedPatientDemo.dob,
+      genderAtBirth: normalizedPatientDemo.genderAtBirth,
     },
   });
 
-  const matchingPatients = patients.filter(patient => {
-    // First, check for an ID match - if it's a match, don't bother checking for demo
-    if (
-      demo.personalIdentifiers &&
-      demo.personalIdentifiers.length > 0 &&
-      intersectionWith(patient.data.personalIdentifiers, demo.personalIdentifiers, isEqual).length >
-        0
-    ) {
-      return true;
-    }
+  // Convert patients to proper datatype
+  const blockedPatientsData = blockedPatients.map(p => convertPatientModelToPatientData(p));
 
-    if (isMatchingDemographics(patient.data, demo)) return true;
-  });
-  if (matchingPatients.length === 0) return null;
-  if (matchingPatients.length === 1) return matchingPatients[0];
-
-  const chosenOne = matchingPatients[0];
+  // Match the blocked patients with the normalized patient using the similarity function
+  const matchingPatients = matchPatients(
+    jaroWinklerSimilarity,
```

Comment:
If this function always takes `jaroWinklerSimilarity` then it should probably just be included in that function

Reply:
Ah yah I disagree. I think we will probably have different similarity function or variations of those functions for our different MPI use cases, and I think its helpful to have this framework. For example, from convos with testing partners, it looks like our similarity function should be stricter than it currently is, and we need to modify it. But the current `jaroWinklerSimilarity` works well for out patient matching within the app

Reply:
That makes sense - i was just under the impression that it was always `jaroWinklerSimilarity` but if it can be other functions then thats fine